src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -566,14 +573,36 @@ private JcrResourceBundle createResourceBundle(
      * </ol>
      */
     private Locale getParentLocale(Locale locale) {
-        if (locale.getVariant().length() != 0) {
+        if (locale.getScript().length() != 0 && locale.getVariant().length() != 0) {
+            try {
+                return new Locale.Builder()
+                        .setLanguage(locale.getLanguage())
+                        .setRegion(locale.getCountry())
+                        .setScript(locale.getScript())
+                        .build();
+            } catch (IllformedLocaleException e) {
+                // fallback to previous implementation
```

Comment:
nitpick: Can this even happen? As you construct the parent locale from an already existing and valid locale object, all elements should be valid.

```suggestion
                // should never happen, as all elements already come from a valid locale object
```